commons/src/main/java/io/hyscale/commons/commands/CommandExecutor.java

```diff
@@ -52,149 +48,149 @@
-	 * else Asynchronously copy command output in
-	 * {@link CommandResult#setCommandOutput(String)} waits for the process to
-	 * complete
-	 * 
-	 * @param command
-	 * @param file
-	 * @param dir
-	 * @return {@link CommandResult}
-	 * @throws IOException
-	 * @throws InterruptedException
-	 * @throws HyscaleException
-	 */
-	private static CommandResult _execute(String command, File file, String dir)
-			throws IOException, InterruptedException, HyscaleException {
-		ProcessBuilder processBuilder = new ProcessBuilder();
-		if (StringUtils.isBlank(dir) || dir.equals(".")) {
-			dir = SetupConfig.CURRENT_WORKING_DIR;
-		}
-		logger.debug("Executing command in dir {}", dir);
-		processBuilder.command(command.split(" "));
-		processBuilder.redirectErrorStream(true);
-		if (file != null) {
-			HyscaleFilesUtil.createEmptyFile(file);
-			processBuilder.redirectOutput(file);
-		}
-		Process process = processBuilder.start();
-		boolean readOutput = false;
-		StringWriter strWriter = new StringWriter();
-		if (file == null) {
-			readOutput = copyOutput(process, strWriter);
-		}
-		CommandResult cmdResult = new CommandResult();
-		int exitCode = 1;
-		try {
-			exitCode = process.waitFor();
-		} catch (InterruptedException e) {
-			logger.error("Error while waiting for process to complete");
-			throw e;
-		}
-		if (readOutput) {
-			cmdResult.setCommandOutput(strWriter.toString());
-		}
-		cmdResult.setExitCode(exitCode);
-		return cmdResult;
-	}
+    /**
+     * Executes command in the directory specified, uses current directory if not
+     * specified If file provided directs output to file (creates if does not exist)
+     * else Asynchronously copy command output in
+     * {@link CommandResult#setCommandOutput(String)} waits for the process to
+     * complete
+     *
+     * @param command
+     * @param file
+     * @param dir
+     * @return {@link CommandResult}
+     * @throws IOException
+     * @throws InterruptedException
+     * @throws HyscaleException
+     */
+    private static CommandResult _execute(String command, File file, String dir,String stdInput)
+            throws IOException, InterruptedException, HyscaleException {
+        ProcessBuilder processBuilder = new ProcessBuilder();
+        if (StringUtils.isBlank(dir) || dir.equals(".")) {
+            dir = SetupConfig.CURRENT_WORKING_DIR;
+        }
+        logger.debug("Executing command in dir {}", dir);
+        processBuilder.command(command.split(" "));
+        processBuilder.redirectErrorStream(true);
+        if (file != null) {
+            HyscaleFilesUtil.createEmptyFile(file);
+            processBuilder.redirectOutput(file);
+        }
+        Process process = processBuilder.start();
+        if(!StringUtils.isBlank(stdInput)) {
+            try (OutputStream processStdin = process.getOutputStream()) {
+                processStdin.write(stdInput.getBytes(StandardCharsets.UTF_8));
+            } catch (IOException e) {
+                throw e;
```

Comment:
Wrap the exception with error code and relevant message and throw HyscaleException
Also add loggers to these kind of errors for more debugging insights .